api/v1alpha1/storagesystem_types.go

```diff
@@ -51,12 +51,13 @@ type StorageSystemSpec struct {
 	// Name describes the name of managed storage vendor CR
 	Name string `json:"name,omitempty"`
 
-	//+kubebuilder:validation:Optional
+	//+kubebuilder:validation:Required
 
 	// NameSpace describes the namespace of managed storage vendor CR
 	NameSpace string `json:"nameSpace,omitempty"`
```

Comment:
If we're saying this field is required, then I would think we shouldn't have `omitempty`? Similar for `Name` above?

Reply:
This is mainly done to make API's backward compatible. As we are adding `Required` tag no one will be able to create a CR without the values. But in the code, we can skip those values if required. Let me know what do you think.

Reply:
Huh... backwards compatible with what? Does this affect the API code generated in such a way that it would impact how we can interact with it?

Reply:
The first answer from this post should help. However, it is a protobuf related post but the problem I want to explain is the same.

https://stackoverflow.com/questions/31801257/why-required-and-optional-is-removed-in-protocol-buffers-3

Reply:
The entire post is talking about why required was *removed* and why it's *bad*. Why are you adding it here?

Reply:
So with this post what I want to show is that protobuf has removed this functionality and now generate all go code with the `omitempty` and I also want to follow the same thing.

Reply:
None of that is clear, `omitempty` is not mentioned anywhere in that post. I still don't understand how this is relevant.

Reply:
Let's discuss this in some of our mtg. I will remove omitempty. If we feel to add we can always do it.